Mathlib/Data/Real/CauSeq.lean

```diff
@@ -0,0 +1,1011 @@
+  have a0 := K0.trans_le ha
+  have b0 := K0.trans_le hb
+  rw [inv_sub_inv' ((abv_pos abv).1 a0) ((abv_pos abv).1 b0), abv_mul abv, abv_mul abv, abv_inv abv,
+    abv_inv abv, abv_sub abv]
+  refine' lt_of_mul_lt_mul_left (lt_of_mul_lt_mul_right _ b0.le) a0.le
+  rw [mul_assoc, inv_mul_cancel_right₀ b0.ne', ← mul_assoc, mul_inv_cancel a0.ne', one_mul]
+  refine' h.trans_le _
+  exact mul_le_mul (mul_le_mul ha le_rfl ε0.le a0.le) hb K0.le (mul_nonneg a0.le ε0.le)
+#align rat_inv_continuous_lemma rat_inv_continuous_lemma
+
+end
+
+/-- A sequence is Cauchy if the distance between its entries tends to zero. -/
+def IsCauSeq {α : Type _} [LinearOrderedField α] {β : Type _} [Ring β] (abv : β → α) (f : ℕ → β) :
+    Prop :=
+  ∀ ε > 0, ∃ i, ∀ j ≥ i, abv (f j - f i) < ε
+#align is_cau_seq IsCauSeq
+
+namespace IsCauSeq
+
+variable [LinearOrderedField α] [Ring β] {abv : β → α} [IsAbsoluteValue abv] {f g : ℕ → β}
+
+-- see Note [nolint_ge]
+--@[nolint ge_or_gt] -- Porting note: restore attribute
+theorem cauchy₂ (hf : IsCauSeq abv f) {ε : α} (ε0 : 0 < ε) :
+    ∃ i, ∀ (j) (_ : j ≥ i) (k) (_ : k ≥ i), abv (f j - f k) < ε := by
+  refine' (hf _ (half_pos ε0)).imp fun i hi j ij k ik => _
+  rw [← add_halves ε]
+  refine' lt_of_le_of_lt (abv_sub_le abv _ _ _) (add_lt_add (hi _ ij) _)
+  rw [abv_sub abv]; exact hi _ ik
+#align is_cau_seq.cauchy₂ IsCauSeq.cauchy₂
+
+theorem cauchy₃ (hf : IsCauSeq abv f) {ε : α} (ε0 : 0 < ε) :
+    ∃ i, ∀ j ≥ i, ∀ k ≥ j, abv (f k - f j) < ε :=
+  let ⟨i, H⟩ := hf.cauchy₂ ε0
+  ⟨i, fun _ ij _ jk => H _ (le_trans ij jk) _ ij⟩
+#align is_cau_seq.cauchy₃ IsCauSeq.cauchy₃
+
+theorem add (hf : IsCauSeq abv f) (hg : IsCauSeq abv g) : IsCauSeq abv (f + g) := fun _ ε0 =>
+  let ⟨_, δ0, Hδ⟩ := rat_add_continuous_lemma abv ε0
+  let ⟨i, H⟩ := exists_forall_ge_and (hf.cauchy₃ δ0) (hg.cauchy₃ δ0)
+  ⟨i, fun _ ij =>
+    let ⟨H₁, H₂⟩ := H _ le_rfl
+    Hδ (H₁ _ ij) (H₂ _ ij)⟩
+#align is_cau_seq.add IsCauSeq.add
+
+end IsCauSeq
+
+/-- `CauSeq β abv` is the type of `β`-valued Cauchy sequences, with respect to the absolute value
+function `abv`. -/
+def CauSeq {α : Type _} [LinearOrderedField α] (β : Type _) [Ring β] (abv : β → α) : Type _ :=
+  { f : ℕ → β // IsCauSeq abv f }
+#align cau_seq CauSeq
+
+namespace CauSeq
+
+variable [LinearOrderedField α]
+
+section Ring
+
+variable [Ring β] {abv : β → α}
+
+instance : CoeFun (CauSeq β abv) fun _ => ℕ → β :=
+  ⟨Subtype.val⟩
+
+-- Porting note: Remove coeFn theorem
+/-@[simp]
+theorem mk_to_fun (f) (hf : IsCauSeq abv f) : @coeFn (CauSeq β abv) _ _ ⟨f, hf⟩ = f :=
+  rfl -/
+#noalign cau_seq.mk_to_fun
+
+theorem ext {f g : CauSeq β abv} (h : ∀ i, f i = g i) : f = g :=
+  Subtype.eq (funext h)
+#align cau_seq.ext CauSeq.ext
+
+theorem is_cau (f : CauSeq β abv) : IsCauSeq abv f :=
+  f.2
+#align cau_seq.is_cau CauSeq.is_cau
+
+theorem cauchy (f : CauSeq β abv) : ∀ {ε}, 0 < ε → ∃ i, ∀ j ≥ i, abv (f j - f i) < ε := by
+  have := f.2
+  rwa [IsCauSeq] at this
```

Comment:
Did the [mathport output](https://github.com/leanprover-community/mathlib4/compare/154f76b355130e6178b0322a7b682eff1f9e0cba..mcdoll/data_real_causeq#diff-18f1434dcab7d38fad2c5144ab5f653c1eef55b9d1d49d20951e556af4ad0317L166-R161) not work here?

Reply:
It does not work.

Reply:
but `@f.2` does